pkg/query/api/v1.go

```diff
@@ -614,3 +616,35 @@ func parseDuration(s string) (time.Duration, error) {
 	}
 	return 0, fmt.Errorf("cannot parse %q to a valid duration", s)
 }
+
+func (api *API) labelNames(r *http.Request) (interface{}, []error, *ApiError) {
+	ctx := r.Context()
+
+	enablePartialResponse, apiErr := api.parsePartialResponseParam(r)
```

Comment:
```suggestion
	enablePartialResponse, err := api.parsePartialResponseParam(r)
```

Reply:
Maybe we could not change to `err`, because it is type of `*ApiError`.